launcher/icons/IconList.cpp

```diff
@@ -337,10 +337,16 @@ void IconList::installIcon(const QString& file, const QString& name)
     if (!fileinfo.isReadable() || !fileinfo.isFile())
         return;
 
-    if (!IconUtils::isIconSuffix(fileinfo.suffix()))
+    const QString sourceSuffix = fileinfo.suffix();
+    if (!IconUtils::isIconSuffix(sourceSuffix))
         return;
 
-    QString target = FS::PathCombine(getDirectory(), name.isEmpty() ? fileinfo.fileName() : name);
+    const QFileInfo targetInfo = name.isEmpty() ? fileinfo : QFileInfo{ name };
+    const QString targetSuffix = targetInfo.suffix();
+    const QString suffix = IconUtils::isIconSuffix(targetSuffix) ? targetSuffix : sourceSuffix;
+
+    const QString targetName = targetInfo.completeBaseName() + QLatin1Char('.') + suffix;
+    QString target = FS::PathCombine(getDirectory(), targetName);
```

Comment:
And then clean up this mess a bit

```suggestion
    const QString targetName = targetInfo.completeBaseName() + QLatin1Char('.') + targetInfo.suffix();
    QString target = FS::PathCombine(getDirectory(), targetName);
```